File: src/game/entity/models/Explosive.java
package game.entity.models;

import java.util.List;
import java.util.Set;

public interface Explosive {
    Set<Class<? extends Entity>> getExplosionObstacles();
    boolean isObstacleOfExplosion(Entity e);
    List<Class<? extends Entity>> getExplosionInteractionEntities();
    int getMaxExplosionDistance();
}

File: src/game/entity/models/Explosive.kt
package game.entity.models

interface Explosive {
    val explosionObstacles: Set<Class<out Entity>>
    fun isObstacleOfExplosion(e: Entity?): Boolean {
        return e == null || explosionObstacles.stream().anyMatch { c: Class<out Entity> -> c.isInstance(e) }
    }

    val explosionInteractionEntities: Set<Class<out Entity>>
    val maxExplosionDistance: Int
}